### ISSUE TITLE
Issue with program

### ISSUE THREAD

**Issue by senyberg** [senyberg/MMAI#5]
Hi @ech2,

Could you have a look at the new code I pushed to our git. There is no reverb yet, working on it, but the program can now read a file and write to a file. Please note that I wrote this from scratch with very limited knowledge of how the libraries work, so there might be unnecessary code in there as well.

There seems to be an issue with the output file though. It is exactly the same as the input. And the file that plays real time sounds awful. If I remember correctly another group had a similar issue, but I cannot seem to find what's wrong with our code. Thanks!

//Sebastian

**Comment by ghost** [comment]
Hi @senyberg,

What exactly should I do with the code? Describe steps to reproduce your issue, please.

Ech

**Comment by senyberg** [comment]
@ech2 you need main.c, portaudio.h, sndfile.h and a wav file in same folder. And then run:
gcc main.c -o main -lsndfile -lportaudio
./main nameoifinput.wav nameofoutput.wav

The issue is the sound it plays while program running (getting some errors). The output file is the same, but no reverb applied so should be fine.

**Comment by senyberg** [comment]
I added the pacman.wav file I have been testing with.

**Comment by ghost** [comment]
@senyberg sorry, I'm a bit busy right now and don't have much time to debug your program. I suggest you to put the `PortAudio` aside until the MVP where it'd be actually needed. If I remember correctly, real-time isn't required for the Monday meet-up. So, to catch up with the milestone schedule, I suggest you to focus on the reverberation now. Probably, you can parallelize the work on the course project in your group, so someone would be working on the `PortAudio` integration, etc.

**Comment by senyberg** [comment]
@ech2 hey again. I have tried to come up with a working offline (using libsndfile only) algorithm for the reverb. But I am running into some problems I cannot seem to understand or find answers to.

For some reason I need to pass a double pointer as argument to the reverb function for it to be able to write something on the new buffer (outbuffer). Right now we have (float *buffer, float **outbuffer). To simply copy the original file to our outbuffer you can do it with *outbuffer = &buffer. But if I try the same but with passing only a single pointer (float *buffer, float*outbuffer) and outbuffer = outbuffer, it will not write anything. 

I understand you cannot debug our whole code, but if you have any tips that could help us get this to work that would be great :)

**Comment by ghost** [comment]
Yes, this is a valid behaviour. Arrays in C/C++ are just pointers, so to
pass this array to function for modification, you need to pass a
*reference* of this array (which is itself a pointer). You need this,
because when you pass a pointer to a function it gets copied, I'd
suggest you to find some info about copying of a pointer in functions.


On Sun, Apr 23, 2017, at 18:42, Sebastian Nyberg wrote:
> @ech2[1] hey again. I have tried to come up with a working offline
> (using libsndfile only) algorithm for the reverb. But I am running
> into some problems I cannot seem to understand or find answers to.> For some reason I need to pass a double pointer as argument to the
> reverb function for it to be able to write something on the new buffer
> (outbuffer). Right now we have (float *buffer, float **outbuffer). To
> simply copy the original file to our outbuffer you can do it with
> *outbuffer = &buffer. But if I try the same but with passing only a
> single pointer (float *buffer, float*outbuffer) and outbuffer =
> outbuffer, it will not write anything.> I understand you cannot debug our whole code, but if you know the
> answer to this it would probably help a lot :)> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].> 


Links:

  1. https://github.com/ech2
  2. https://github.com/senyberg/MMAI/issues/5#issuecomment-296452084
